core/Plugin/Archiver.php

```diff
@@ -164,13 +165,30 @@ private function getRecordBuilders(string $pluginName): array
          */
         Piwik::postEvent('Archiver.filterRecordBuilders', [&$recordBuilders]);
 
+        return $recordBuilders;
+    }
+
+    private function filterRecordBuildersByRequestedRecords(array $recordBuilders): array
```

Comment:
Shouldn't we pass in `$requestedReports` as a parameter here? To remove the ambiguity of the the function.